stable_baselines3/common/base_class.py

```diff
@@ -529,6 +529,7 @@ def set_env(self, env: GymEnv, force_reset: bool = True) -> None:
 
         self.n_envs = env.num_envs
         self.env = env
+        self._setup_model()
```

Comment:
this is actually too aggressive, we should reset buffer and only if the number of envs is different, otherwise we lose the trained parameters of the policy/vf (setup_model reinitialize the policy too).
And in the case of replay buffer, we should probably have a warning because all data stored will be lost.

Reply:
or best is to forbid using `set_env()` with a different number of envs and force users to use `load()` in that case

Reply:
As I re-read my code, I realize that this change is outside the scope of the PR.